packages/inertia-vue/src/form.js

```diff
@@ -0,0 +1,108 @@
+import Vue from 'vue'
+import { Inertia } from '@inertiajs/inertia'
+
+export default function(data) {
+  const defaults = JSON.parse(JSON.stringify(data))
+
+  return {
+    errors: {},
+    hasErrors: false,
+    processing: false,
+    data() {
+      return Object
+        .keys(data)
+        .reduce((carry, key) => {
+          carry[key] = this[key]
+          return carry
+        }, {})
+    },
+    submit(method, url, options) {
+      url = url || Inertia.page.url
+      options = options || {}
+
+      Inertia[method](url, this.data(), {
+        ... options,
+        onStart: visit => {
+          this.processing = true
+
+          if (options.onStart) {
+            return options.onStart(visit)
+          }
+        },
+        onFinish: () => {
+          this.processing = false
+
+          if (options.onFinish) {
+            return options.onFinish()
+          }
+        },
+        onError: errors => {
+          Vue.set(this, 'errors', errors)
+          this.hasErrors = true
+
+          if (options.onError) {
+            return options.onError(errors)
+          }
+        },
+        onSuccess: page => {
+          Vue.set(this, 'errors', {})
+          this.hasErrors = false
+
+          if (options.onSuccess) {
+            return options.onSuccess(page)
+          }
+        },
+      })
+    },
+    post(url, options) {
+      this.submit('post', url, options)
+    },
+    put(url, options) {
+      this.submit('put', url, options)
+    },
+    patch(url, options) {
+      this.submit('patch', url, options)
+    },
+    delete(url, options) {
+      this.submit('delete', url, options)
+    },
+    reset(fields) {
+      const args = Array.isArray(fields)
+        ? fields
+        : Object.values(arguments)
+
+      if (args.length === 0) {
+        Object.assign(this, defaults)
+      } else {
+        Object.assign(
+          this,
+          Object
+            .keys(defaults)
+            .filter(key => args.includes(key))
+            .reduce((carry, key) => {
+              carry[key] = defaults[key]
+              return carry
+            }, {}),
+        )
+      }
+
+      return this.clearErrors()
+    },
+    clearErrors() {
+      this.errors = {}
```

Comment:
I think this should reset `hasErrors`.

```suggestion
      this.errors = {}
      this.hasErrors = false
```

Not sure whether this needs to use `Vue.set` (You used it in `onSuccess`)
```suggestion
      Vue.set(this, 'errors', {})
      this.hasErrors = false
```